cpu/nrf52/ldscripts/nrf52832xxaa.ld

```diff
@@ -18,10 +18,23 @@
  * @}
  */
 
+rom_length = 512K;
+
+INCLUDE multislot.ld
+
 MEMORY
 {
-    rom (rx)    : ORIGIN = 0x00000000, LENGTH = 512K
-    ram (rwx)   : ORIGIN = 0x20000000, LENGTH = 64K
+    rom (rx)    : ORIGIN = 0x00000000 + boot_offset, LENGTH = rom_length
+    ram (rwx)   : ORIGIN = 0x20000000,               LENGTH = 64K
+}
+
+SECTIONS
+{
+    /* Image headers for MCUBoot */
+    .imghdr (NOLOAD):
+    {
+      . = . + img_hdr;
+    } > rom
```

Comment:
> I think keeping this variable in the linker script doesn't harm

It needs adaption of the linker scripts

> and adds readability when looking at the ELF file where we can find the specific allocated space for the headers.

true, but ...

> This can be useful when using different tools to add these headers directly to the ELF file, instead if padding them to the binary file. 

This is my most concern. If the header is part of the image, the complexity of signing images increases a lot. How's the signature generated? Is the header part of it? We'd also basically force usage of an ELF aware image signing tool, instead of simply cat'ing header and elf together.

> It also helps to know while debugging where the real image starts, while also knowing why the vector table is located 512B after the bootloader.

- the debugger knows the image start regardless of the image hdr defined in the linker address
- the debugger doesn't mention the image header anywhere, regardless of it being defined in the linker script or not
- if the image is compiled with knowledge about MCUBOOT and it's headers, it can deduce the image hdr start, and knows it's length

I'm not convinced. The only reason to put this into the linker script is to give the linker a hint about the size of the image header, which can also be implicitly given by setting a correct offset. On the other hand, this adds to maintanance and complexity. so please remove.